classes/checkout/CheckoutPaymentStep.php

```diff
@@ -84,7 +84,10 @@ public function render(array $extraParams = [])
         } else {
             $selectedDeliveryOption = 0;
         }
-        unset($selectedDeliveryOption['product_list']);
+
+        if (true === is_array($selectedDeliveryOption)) {
+            unset($selectedDeliveryOption['product_list']);
```

Comment:
Just a question, does it mean sometimes `$selectedDeliveryOption` is an array, sometimes it's not? Maybe that's an issue no? Seems like you're patching the symptom but the root cause remains, there is some inconsistency on this variable that is weird.

Starting with the previous line which sets the fallback value to 0, not sure it makes much sense.. Or maybe the unset should simply be moved into the `if (isset($deliveryOptions[$deliveryOptionKey])) {` block

Reply:
In the case I tested, `$selectedDeliveryOption` was null.

Reply:
@nicosomb, how is that? it's either value of `$deliveryOptions[$deliveryOptionKey]` or `0` 🤔 and there's `isset($deliveryOptions[$deliveryOptionKey]` above

+ there's a check to see if it's an array, but there's no check if the key exists, maybe something like:

`is_array($selectedDeliveryOption) && isset($selectedDeliveryOption['product_list'])` would be more precise?